### DISCUSSION TITLE
My old X11 app `pmck` (poor man's clock)

### DISCUSSION THREAD

**Discussion by 01micko** [labwc/labwc#2212] [Q&A]
[pmck repo](https://github.com/01micko/pmck)

I wrote this approximately 120 moons ago and thought that bitrot may have killed it but no; compiles without warnings and runs fine, even as Xwayland with some 'buts'.

It's written with `cairo` using the internal toy font API. Font's are only shown in 2 of the 4 options any way (read the config file to get a grasp).

It still works perfectly on XFCE (I tested on many DEs /WMs at the time - which led to the builtin library to find the right desktop.)

On labwc latest master it works - also works in 0.7.4 on debian testing.

BUT - `override-redirect` doesn't seem to work. So there is a black square around it and it's on the top layer.

### XFCE
![xfce](https://github.com/user-attachments/assets/d92c3506-2655-43fb-bd77-bb71665e226c)

### Labwc
![20241004_15h42m17s_grim](https://github.com/user-attachments/assets/d06cdc47-dd4f-4fa9-bb22-3fa691f0ec31)

Now since it kind of works, shouldn't Xwayland respect override-redirect?

TBH I don't really care that it doesn't work, more curious as to why. I suppose I should grab some debug logs from latest-master to look for clues. 

At some stage I intend re-write it with `gtk+-3.0/gtk-layer-shell` or dive into `egl` like I dove into the internals of `X`!

**Comment by stefonarch** [comment]
FYI @tsujan had done something called "glassyclock" (no repo afaik for it):

![image](https://github.com/user-attachments/assets/ea0a5ba9-0de9-490f-9c88-69ec6b524de1)

**Reply by 01micko** [reply]
looks nice :)

**Reply by tsujan** [reply]
Mine is just an SVG-based, Qt-based analog clock, tuned to KWin and its blur effect and working anywhere under X11 and Wayland. I didn't put it in my GitHub page because I didn't have time to accept any feature request. May upload it without enabling feature requests (if possible).

![clock](https://github.com/user-attachments/assets/3ccdcc28-38f5-4ed6-a30c-411645c9ddd9)

**Reply by 01micko** [reply]
If it's svg just post a `gist`

**Comment by 01micko** [comment]
Anyway, I ended up getting analog clocks to work in `sfwbar` with `svg`.

https://github.com/01micko/sfwbar-configs/blob/main/configs/sfwbar/world-clock/Readme-world-clock.md
